### PR TITLE
Core/Movement: Fix wrong orientation set when pausing movement

### LINKED ISSUE #23272: Creature/Movement: Orientation changes wrongly and not updated in client side
**Description:** When a npc use a waypoints via SAI then it pause in some points, orientation will change to the opposite without being updated visually in client side.
Example : Npc: 16708
He has a waypoint: 16708 that start via SAI and stops in 2 spots to execute some cosmetics:
Dekin - ON Waypoint 6 Reached - Run Script
Dekin - ON Waypoint 12 Reached - Run Script

When the npc reach point 6 the orientation is 5.179513 but upon paused it changes to 1.985356

When the npc reach point 12 the orientation is : Orientation: 4.167312 but upon paused it changes to 0.886503

If you move from the creature until it disappear then come back the orientation will get updated for the client

**Steps to reproduce the problem:**

1. .go c id 16708 
2. wait until the npc stops in one oh the 2 spots
3. .gps the npc and compare the orientation

Orientation not updated in client side
image

Orientation updated in client side
image

**Branch(es):** 3.3.5

**TC rev. hash/commit:**  rev. cf37f357df5c

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Fix wrong orientation set when pausing movement. The current Unit position is set as calculated from the spline if no time has passed since the spline was launched, i.e. time_passed == 0

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #23272

**Tests performed:** (Does it build, tested in-game, etc.)
- Tested ingame and debugged to see that the orientation of the npc of #23272 https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=16708 is correct
- Talked to the npc of #23272 https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=16708 while it's walking, the position got updated correctly (even if there is another issue that the old orientation is set when talking to the npc, relogging shows the correct orientation, it should be checked if 335 had this issue already).

**Known issues and TODO list:** (add/remove lines as needed)

-  None

### CODE REVIEW

**Comment by jackpoz** [comment]
@ccrs @Shauren please take a look at this

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -10171,7 +10171,8 @@ void Unit::StopMoving()
         return;
 
     // Update position now since Stop does not start a new movement that can be updated later
-    UpdateSplinePosition();
+    if (movespline->Started())
```
Comment (Ovahlord) [inline src/server/game/Entities/Unit/Unit.cpp]: isn't !movespline->Finalized()  the common helper used for checking if a spline is currently in progress?

Comment (jackpoz) [reply src/server/game/Entities/Unit/Unit.cpp]: Does that tell if the spline started yet ?

Comment (Ovahlord) [reply src/server/game/Entities/Unit/Unit.cpp]: It does return true as long as a spline is currently in effect, yes. It's the common check used in every movement generator to check if a creature is actually moving or not. (It's also the common check to protect against nullptrs: no spline in effect --> movespline will return nullptr)

Comment (jackpoz) [reply src/server/game/Entities/Unit/Unit.cpp]: But I don't need to know if it's in effect or finished,  I need to know if the Unit walked at least 1 step already in three current spline (hence the name Started and not something like InProgress)

**Comment by jackpoz** [comment]
@Shauren anything against merging this ?

**Pull request merged**